sphinx/writers/latex.py

```diff
@@ -1333,15 +1340,27 @@ def depart_tgroup(self, node):
 
     def visit_thead(self, node):
         # type: (nodes.Node) -> None
-        self.pushbody(self.table.header)  # Redirect head output until header is finished.
+        # Redirect head output until header is finished.
+        self.pushbody(self.table.header)
+        # footnotes in longtable header must be restricted
+        self.restrict_footnote(node)
 
     def depart_thead(self, node):
         # type: (nodes.Node) -> None
         self.popbody()
+        # temporary buffer for footnotes from table header
+        self.pushbody([])
+        self.unrestrict_footnote(node)
+        # the footnote texts from header
+        self.table.header_footnotetexts = self.popbody()
 
     def visit_tbody(self, node):
         # type: (nodes.Node) -> None
-        self.pushbody(self.table.body)  # Redirect body output until table is finished.
+        # Redirect body output until table is finished.
+        self.pushbody(self.table.body)
+        # insert footnotetexts from header at start of body (due to longtable)
+        # those from caption are handled by templates (to allow caption at foot)
+        self.body.extend(self.table.header_footnotetexts)
```

Comment:
Yes, a LaTeX test is needed in my opinion but if the case of footnotetexts from caption is treated it will probably be also test for header_footnotetexts. I don't think we need pure Python test on header_footnotetexts, only check that it is well inserted at start of table body. I will check.